backend/src/utils/setup/backfillData.ts

```diff
@@ -679,22 +686,79 @@ export const backfillUserAuthMethods = async () => {
 };
 
 export const backfillPermission = async () => {
-  await Membership.updateMany(
-    {
-      deniedPermissions: {
-        $exists: true,
-        $ne: []
-      },
-      role: MEMBER
+  const memberships = await Membership.find({
+    deniedPermissions: {
+      $exists: true,
+      $ne: []
     },
-    [
-      {
-        $set: {
-          role: VIEWER
-        }
+    role: MEMBER
+  })
+    .populate<{ workspace: IWorkspace }>("workspace")
+    .lean();
+
+  for (const membership of memberships) {
+    // get permissions of members except secret permission
+    const customPermissions = memberProjectPermissions.rules.filter(
+      ({ subject }) => subject !== ProjectPermissionSub.Secrets
+    );
+    const secretAccessRule: Record<string, { read: boolean; write: boolean }> = {};
+
+    // iterate and record true and false ones
+    membership.deniedPermissions.forEach(({ ability, environmentSlug }) => {
+      if (!secretAccessRule?.[environmentSlug])
+        secretAccessRule[environmentSlug] = { read: true, write: true };
+      if (ability === "write") secretAccessRule[environmentSlug].write = false;
+      if (ability === "read") secretAccessRule[environmentSlug].read = false;
+    });
+
+    Object.entries(secretAccessRule).forEach(([envSlug, { read, write }]) => {
+      if (read) {
+        customPermissions.push({
+          subject: ProjectPermissionSub.Secrets,
+          action: ProjectPermissionActions.Read,
+          conditions: { environment: envSlug }
```

Comment:
we should also set the secrets path to `/**` so users have access to eveything within that environment just like they would right now